cmd/crank/render/render.go

```diff
@@ -236,13 +237,14 @@ func Render(ctx context.Context, log logging.Logger, in Inputs) (Outputs, error)
 	// Load user-supplied context.
 	for k, data := range in.Context {
 		var jv any
-		if err := json.Unmarshal(data, &jv); err != nil {
-			return Outputs{}, errors.Wrapf(err, "cannot unmarshal JSON value for context key %q", k)
+
+		if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), 4096).Decode(&jv); err != nil {
```

Comment:
Doesn't look like we're using `NewYAMLOrJSONDecoder` anywhere else in Crossplane today. The standard `Unmarshal` methods in `k8s.io/apimachinery/pkg/util/yaml` and `sigs.k8s.io/yaml` both handle JSON correctly, and are more commonly used in Crossplane and the ecosystem.

Any reason to prefer explicitly creating a `YAMLOrJSONDecoder`?